src/Temporalio/Worker/Interceptors/CreateContinueAsNewExceptionInput.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections.Generic;
+using Temporalio.Api.Common.V1;
+using Temporalio.Workflows;
+
+namespace Temporalio.Worker.Interceptors
+{
+    /// <summary>
+    /// Input for <see cref="WorkflowOutboundInterceptor.CreateContinueAsNewException" />.
+    /// </summary>
+    /// <param name="Workflow">Workflow to continue.</param>
+    /// <param name="Args">Arguments for the workflow.</param>
+    /// <param name="Options">Options passed in to continue as new.</param>
+    /// <param name="Headers">Headers to include.</param>
+    public record CreateContinueAsNewExceptionInput(
+        string Workflow,
```

Comment:
Could potentially be useful to pass a reference to the workflow class if you have it so the interceptor can extract the argument types.
Same goes for the activity input.